// @twilio-paste/theme/dist/index.debug.es.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/themeProvider.tsx
import * as React from "react";
import { useReducedMotion, Globals as AnimatedGlobals } from "@twilio-paste/animation-library";
import { styled, StylingGlobals, ThemeProvider as StyledThemeProvider } from "@twilio-paste/styling-library";

// src/themes/default/index.ts
import {
  backgroundColors,
  borderColors,
  borderWidths,
  radii,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  boxShadows,
  sizings,
  spacings,
  textColors,
  zIndices,
  colors
} from "@twilio-paste/design-tokens";

// src/generateThemeFromTokens.ts
var generateThemeFromTokens = ({
  backgroundColors: backgroundColors4,
  borderColors: borderColors4,
  borderWidths: borderWidths4,
  radii: radii4,
  fonts: fonts4,
  fontSizes: fontSizes4,
  fontWeights: fontWeights4,
  lineHeights: lineHeights4,
  boxShadows: boxShadows4,
  sizings: sizings4,
  spacings: spacings4,
  textColors: textColors4,
  zIndices: zIndices4,
  colors: colors4
}) => {
  const breakpoints = [sizings4.size40, sizings4.size100, sizings4.size120];
  const dataVisualization = Object.keys(colors4).filter((colorToken) => {
    return colorToken.includes("DataVisualization");
  }).reduce((obj, key) => __spreadProps(__spreadValues({}, obj), {
    [key]: colors4[key]
  }), {});
  return {
    shadows: boxShadows4,
    borderWidths: borderWidths4,
    radii: radii4,
    breakpoints,
    textColors: textColors4,
    borderColors: borderColors4,
    backgroundColors: backgroundColors4,
    fonts: fonts4,
    fontSizes: fontSizes4,
    fontWeights: fontWeights4,
    lineHeights: lineHeights4,
    widths: sizings4,
    maxWidths: sizings4,
    minWidths: sizings4,
    heights: sizings4,
    maxHeights: sizings4,
    minHeights: sizings4,
    sizes: sizings4,
    iconSizes: {
      sizeIcon10: sizings4.sizeIcon10,
      sizeIcon20: sizings4.sizeIcon20,
      sizeIcon30: sizings4.sizeIcon30,
      sizeIcon40: sizings4.sizeIcon40,
      sizeIcon50: sizings4.sizeIcon50,
      sizeIcon60: sizings4.sizeIcon60,
      sizeIcon70: sizings4.sizeIcon70,
      sizeIcon80: sizings4.sizeIcon80,
      sizeIcon90: sizings4.sizeIcon90,
      sizeIcon100: sizings4.sizeIcon100,
      sizeIcon110: sizings4.sizeIcon110
    },
    space: spacings4,
    zIndices: zIndices4,
    dataVisualization
  };
};

// src/themes/default/index.ts
var DefaultTheme = generateThemeFromTokens({
  backgroundColors,
  borderColors,
  borderWidths,
  radii,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  boxShadows,
  sizings,
  spacings,
  textColors,
  zIndices,
  colors
});

// src/themes/sendgrid/index.ts
import {
  backgroundColors as backgroundColors2,
  borderColors as borderColors2,
  borderWidths as borderWidths2,
  radii as radii2,
  fonts as fonts2,
  fontSizes as fontSizes2,
  fontWeights as fontWeights2,
  lineHeights as lineHeights2,
  boxShadows as boxShadows2,
  sizings as sizings2,
  spacings as spacings2,
  textColors as textColors2,
  zIndices as zIndices2,
  colors as colors2
} from "@twilio-paste/design-tokens/dist/themes/sendgrid/tokens.es6";
var SendGridTheme = generateThemeFromTokens({
  backgroundColors: backgroundColors2,
  borderColors: borderColors2,
  borderWidths: borderWidths2,
  radii: radii2,
  fonts: fonts2,
  fontSizes: fontSizes2,
  fontWeights: fontWeights2,
  lineHeights: lineHeights2,
  boxShadows: boxShadows2,
  sizings: sizings2,
  spacings: spacings2,
  textColors: textColors2,
  zIndices: zIndices2,
  colors: colors2
});

// src/themes/dark/index.ts
import {
  backgroundColors as backgroundColors3,
  borderColors as borderColors3,
  borderWidths as borderWidths3,
  radii as radii3,
  fonts as fonts3,
  fontSizes as fontSizes3,
  fontWeights as fontWeights3,
  lineHeights as lineHeights3,
  boxShadows as boxShadows3,
  sizings as sizings3,
  spacings as spacings3,
  textColors as textColors3,
  zIndices as zIndices3,
  colors as colors3
} from "@twilio-paste/design-tokens/dist/themes/dark/tokens.es6";
var DarkTheme = generateThemeFromTokens({
  backgroundColors: backgroundColors3,
  borderColors: borderColors3,
  borderWidths: borderWidths3,
  radii: radii3,
  fonts: fonts3,
  fontSizes: fontSizes3,
  fontWeights: fontWeights3,
  lineHeights: lineHeights3,
  boxShadows: boxShadows3,
  sizings: sizings3,
  spacings: spacings3,
  textColors: textColors3,
  zIndices: zIndices3,
  colors: colors3
});

// src/themes/index.ts
var ConsoleTheme = DefaultTheme;

// src/styles/global.ts
import { css } from "@twilio-paste/styling-library";
var pasteGlobalStyles = css({
  html: {
    fontSize: "100%",
    fontFamily: "fontFamilyText"
  },
  "html:lang(ja), html:lang(ja-JP)": {
    ".paste-theme-provider": {
      fontFamily: "fontFamilyTextJapanese"
    }
  },
  "html:lang(zh-CN)": {
    ".paste-theme-provider": {
      fontFamily: "fontFamilyTextChineseSimplified"
    }
  },
  "html:lang(zh-TW), html:lang(zh-HK)": {
    ".paste-theme-provider": {
      fontFamily: "fontFamilyTextChineseTraditional"
    }
  },
  "html:lang(ko)": {
    ".paste-theme-provider": {
      fontFamily: "fontFamilyTextKorean"
    }
  },
  body: {
    backgroundColor: "colorBackgroundBody",
    margin: 0,
    fontSize: "fontSize30"
  },
  "*, *::after, *::before": {
    boxSizing: "border-box"
  },
  "@media (prefers-reduced-motion: reduce)": {
    "*": {
      animationDuration: "0 !important",
      animationIterationCount: "1 !important",
      transitionDuration: "0 !important",
      scrollBehavior: "auto !important"
    }
  },
  ":root": {
    ["--reach-dialog"]: "1"
  }
});

// src/styles/base.ts
import { css as css2 } from "@twilio-paste/styling-library";
var pasteBaseStyles = css2({
  color: "colorText",
  fontSize: "fontSize30",
  fontFamily: "fontFamilyText",
  lineHeight: "lineHeight30",
  fontWeight: "fontWeightNormal",
  fontVariantNumeric: "tabular-nums"
});

// src/styles/fonts.ts
import { EmotionCSS } from "@twilio-paste/styling-library";
var pasteFonts = EmotionCSS`
/* -------------------------------------------------------
Variable font.
*/
@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: normal;
  font-named-instance: 'Regular';
  src: url("https://assets.twilio.com/public_assets/paste-fonts/main-1.2.0/Inter-roman.var.woff2?v=3.15") format("woff2");
}
@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: italic;
  font-named-instance: 'Italic';
  src: url("https://assets.twilio.com/public_assets/paste-fonts/main-1.2.0/Inter-italic.var.woff2?v=3.15") format("woff2");
}
/* --------------------------------------------------------------------------
[EXPERIMENTAL] Multi-axis, single variable font.

Slant axis is not yet widely supported (as of February 2019) and thus this
multi-axis single variable font is opt-in rather than the default.

When using this, you will probably need to set font-variation-settings
explicitly, e.g.

  * { font-variation-settings: "slnt" 0deg }
  .italic { font-variation-settings: "slnt" 10deg }

*/
@font-face {
  font-family: 'Inter var experimental';
  font-weight: 100 900;
  font-display: swap;
  font-style: oblique 0deg 8deg;
  src: url("https://assets.twilio.com/public_assets/paste-fonts/main-1.2.0/Inter.var.woff2?v=3.15") format("woff2");
}

/*
Fira Mono
*/
@font-face {
  font-family: 'Fira Mono';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: local(''),
       url('https://assets.twilio.com/public_assets/paste-fonts/main-1.2.0/fira-mono-v9-latin-ext_latin_greek-ext_greek_cyrillic-ext_cyrillic-regular.woff2') format('woff2'),
       url('https://assets.twilio.com/public_assets/paste-fonts/main-1.2.0/fira-mono-v9-latin-ext_latin_greek-ext_greek_cyrillic-ext_cyrillic-regular.woff') format('woff');
}
@font-face {
  font-family: 'Fira Mono';
  font-style: normal;
  font-display: swap;
  font-weight: 700;
  src: local(''),
       url('https://assets.twilio.com/public_assets/paste-fonts/main-1.2.0/fira-mono-v9-latin-ext_latin_greek-ext_greek_cyrillic-ext_cyrillic-700.woff2') format('woff2'),
       url('https://assets.twilio.com/public_assets/paste-fonts/main-1.2.0/fira-mono-v9-latin-ext_latin_greek-ext_greek_cyrillic-ext_cyrillic-700.woff') format('woff');
}
@font-face {
  font-family: 'Fira Mono';
  font-style: normal;
  font-display: swap;
  font-weight: 500;
  src: local(''),
       url('https://assets.twilio.com/public_assets/paste-fonts/main-1.2.0/fira-mono-v9-latin-ext_latin_greek-ext_greek_cyrillic-ext_cyrillic-500.woff2') format('woff2'),
       url('https://assets.twilio.com/public_assets/paste-fonts/main-1.2.0/fira-mono-v9-latin-ext_latin_greek-ext_greek_cyrillic-ext_cyrillic-500.woff') format('woff');
}
`;

// src/constants.ts
var ThemeVariants = {
  DEFAULT: "default",
  SENDGRID: "sendgrid",
  FLEX: "flex",
  DARK: "dark"
};
var DeprecatedThemeVariants = {
  CONSOLE: "console"
};

// src/utils/isDeprecatedTheme.ts
var isDeprecatedTheme = (theme) => {
  const themeToDeprecationMessage = {
    console: "It was removed in the 2022.02.04 release of Paste."
  };
  console.error(`Paste: ${theme} is a deprecated theme. ${themeToDeprecationMessage[theme]}`);
  return null;
};

// src/themeProvider.tsx
var StyledBase = styled.div(pasteBaseStyles);
function getProviderThemeProps(theme, customBreakpoints) {
  switch (theme) {
    case ThemeVariants.DARK:
      return __spreadProps(__spreadValues({}, DarkTheme), {
        breakpoints: customBreakpoints || DarkTheme.breakpoints
      });
    case ThemeVariants.SENDGRID:
      return __spreadProps(__spreadValues({}, SendGridTheme), {
        breakpoints: customBreakpoints || SendGridTheme.breakpoints
      });
    case DeprecatedThemeVariants.CONSOLE:
      isDeprecatedTheme(DeprecatedThemeVariants.CONSOLE);
      return __spreadProps(__spreadValues({}, DefaultTheme), {
        breakpoints: customBreakpoints || DefaultTheme.breakpoints
      });
    case ThemeVariants.FLEX:
    case ThemeVariants.DEFAULT:
    default:
      return __spreadProps(__spreadValues({}, DefaultTheme), {
        breakpoints: customBreakpoints || DefaultTheme.breakpoints
      });
  }
}
var ThemeProvider = (_a) => {
  var _b = _a, {
    customBreakpoints,
    theme = ThemeVariants.DEFAULT,
    disableAnimations = false
  } = _b, props = __objRest(_b, [
    "customBreakpoints",
    "theme",
    "disableAnimations"
  ]);
  const prefersReducedMotion = useReducedMotion();
  React.useMemo(() => {
    AnimatedGlobals.assign({
      skipAnimation: disableAnimations || prefersReducedMotion
    });
  }, [disableAnimations, prefersReducedMotion]);
  const providerThemeProps = getProviderThemeProps(theme, customBreakpoints);
  return /* @__PURE__ */ React.createElement(StyledThemeProvider, {
    theme: providerThemeProps
  }, /* @__PURE__ */ React.createElement(StylingGlobals, {
    styles: pasteGlobalStyles({ theme: providerThemeProps })
  }), /* @__PURE__ */ React.createElement(StylingGlobals, {
    styles: pasteFonts
  }), /* @__PURE__ */ React.createElement(StyledBase, __spreadValues({
    className: "paste-theme-provider"
  }, props)));
};

// src/themeConsumer.tsx
import * as React2 from "react";
import { ThemeContext } from "@twilio-paste/styling-library";
var ThemeConsumer = (_a) => {
  var _b = _a, { children } = _b, props = __objRest(_b, ["children"]);
  if (children == null || typeof children !== "function") {
    throw new Error("[ThemeConsumer]: You must pass a function as children");
  }
  return /* @__PURE__ */ React2.createElement(ThemeContext.Consumer, null, (theme) => {
    const newTheme = theme;
    return children(__spreadProps(__spreadValues({}, props), { theme: newTheme }));
  });
};

// src/withTheme.ts
import { withTheme } from "@twilio-paste/styling-library";

// src/useTheme.tsx
import * as React3 from "react";
import { ThemeContext as ThemeContext2 } from "@twilio-paste/styling-library";
var useTheme = () => {
  const context = React3.useContext(ThemeContext2);
  if (!context) {
    throw new Error("[useHook]: must be used within the @twilio-paste/theme provider");
  }
  return context;
};

// src/generateTokensFromTheme.ts
var generateTokensFromTheme = ({
  backgroundColors: backgroundColors4,
  borderColors: borderColors4,
  borderWidths: borderWidths4,
  radii: radii4,
  fonts: fonts4,
  fontSizes: fontSizes4,
  fontWeights: fontWeights4,
  lineHeights: lineHeights4,
  shadows,
  sizes,
  space,
  textColors: textColors4,
  zIndices: zIndices4,
  dataVisualization
}) => {
  return {
    boxShadows: shadows,
    borderWidths: borderWidths4,
    radii: radii4,
    textColors: textColors4,
    borderColors: borderColors4,
    backgroundColors: backgroundColors4,
    fonts: fonts4,
    fontSizes: fontSizes4,
    fontWeights: fontWeights4,
    lineHeights: lineHeights4,
    sizings: sizes,
    spacings: space,
    zIndices: zIndices4,
    colors: dataVisualization
  };
};

// src/utils/remToPx.ts
var PX_PER_REM = 16;
var remToPx = (rem, type = "number") => {
  const remValue = typeof rem === "string" ? Number.parseFloat(rem.replace("rem", "")) : rem;
  const pxValue = Math.round(remValue * PX_PER_REM);
  return type === "number" ? pxValue : `${pxValue}px`;
};

// src/useThemeContrastCheck.ts
import {
  getContrastRatingsOfTokensWithTextContrastRequirements,
  getContrastRatingsOfTokensWithUIControlContrastRequirements,
  getNumberOfTextFailures,
  getNumberOfUIControlFailures
} from "@twilio-paste/color-contrast-utils";
var useThemeContrastCheck = () => {
  const theme = useTheme();
  const designTokens = generateTokensFromTheme(theme);
  const textContrastRating = getContrastRatingsOfTokensWithTextContrastRequirements(designTokens);
  const uiControlContrastRating = getContrastRatingsOfTokensWithUIControlContrastRequirements(designTokens);
  const numberOfTextFailures = getNumberOfTextFailures(textContrastRating);
  const numberOfUIControlFailures = getNumberOfUIControlFailures(uiControlContrastRating);
  const totalFailures = numberOfTextFailures + numberOfUIControlFailures;
  return {
    textContrastRating,
    uiControlContrastRating,
    numberOfTextFailures,
    numberOfUIControlFailures,
    totalFailures
  };
};

// src/index.tsx
var Theme = {
  Provider: ThemeProvider,
  Consumer: ThemeConsumer
};
export {
  ConsoleTheme,
  DarkTheme,
  DefaultTheme,
  SendGridTheme,
  StyledBase,
  Theme,
  ThemeVariants,
  generateThemeFromTokens,
  generateTokensFromTheme,
  pasteBaseStyles,
  pasteFonts,
  pasteGlobalStyles,
  remToPx,
  useTheme,
  useThemeContrastCheck,
  withTheme
};
