// @twilio-paste/skeleton-loader/dist/index.debug.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __markAsModule = (target) => __defProp(target, "__esModule", { value: true });
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __reExport = (target, module2, copyDefault, desc) => {
  if (module2 && typeof module2 === "object" || typeof module2 === "function") {
    for (let key of __getOwnPropNames(module2))
      if (!__hasOwnProp.call(target, key) && (copyDefault || key !== "default"))
        __defProp(target, key, { get: () => module2[key], enumerable: !(desc = __getOwnPropDesc(module2, key)) || desc.enumerable });
  }
  return target;
};
var __toESM = (module2, isNodeMode) => {
  return __reExport(__markAsModule(__defProp(module2 != null ? __create(__getProtoOf(module2)) : {}, "default", !isNodeMode && module2 && module2.__esModule ? { get: () => module2.default, enumerable: true } : { value: module2, enumerable: true })), module2);
};
var __toCommonJS = /* @__PURE__ */ ((cache) => {
  return (module2, temp) => {
    return cache && cache.get(module2) || (temp = __reExport(__markAsModule({}), module2, 1), cache && cache.set(module2, temp), temp);
  };
})(typeof WeakMap !== "undefined" ? /* @__PURE__ */ new WeakMap() : 0);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  SkeletonLoader: () => SkeletonLoader
});
var React = __toESM(require("react"));
var import_animation_library = require("@twilio-paste/animation-library");
var import_styling_library = require("@twilio-paste/styling-library");
var import_box = require("@twilio-paste/box");
var AnimatedSkeleton = (0, import_animation_library.animated)(import_box.Box);
var StyledAnimatedSkeleton = (0, import_styling_library.styled)(AnimatedSkeleton)(() => (0, import_styling_library.css)({
  background: `linear-gradient(90deg, transparent, rgba(255, 255, 255, 0.4) 40%, rgba(255, 255, 255, 0.4) 60%, transparent)`,
  transform: `translateX(-100%) skew(155deg)`
}));
var animatedConfig = {
  loop: { delay: 700, reset: true },
  from: { translateX: "-100%", skew: "155deg" },
  to: { translateX: "105%", skew: "155deg" },
  config: {
    mass: 0.1,
    tension: 80,
    friction: 50
  }
};
var SkeletonLoader = React.forwardRef((_a, ref) => {
  var _b = _a, {
    borderBottomLeftRadius,
    borderBottomRightRadius,
    borderRadius = "borderRadius20",
    borderTopLeftRadius,
    borderTopRightRadius,
    element = "SKELETON_LOADER",
    display,
    height = "sizeIcon20",
    maxHeight,
    maxWidth,
    minHeight,
    minWidth,
    size = null,
    width = null
  } = _b, props = __objRest(_b, [
    "borderBottomLeftRadius",
    "borderBottomRightRadius",
    "borderRadius",
    "borderTopLeftRadius",
    "borderTopRightRadius",
    "element",
    "display",
    "height",
    "maxHeight",
    "maxWidth",
    "minHeight",
    "minWidth",
    "size",
    "width"
  ]);
  const animatedSkeletonStyles = (0, import_animation_library.useSpring)(animatedConfig);
  return /* @__PURE__ */ React.createElement(import_box.Box, __spreadProps(__spreadValues({}, (0, import_box.safelySpreadBoxProps)(props)), {
    "aria-busy": "true",
    backgroundColor: "colorBackgroundStrong",
    borderBottomLeftRadius,
    borderBottomRightRadius,
    borderRadius,
    borderTopLeftRadius,
    borderTopRightRadius,
    display,
    element,
    height,
    maxHeight,
    maxWidth,
    minHeight,
    minWidth,
    overflow: "hidden",
    pointerEvents: "none",
    position: "relative",
    userSelect: "none",
    size,
    width,
    ref
  }), /* @__PURE__ */ React.createElement(StyledAnimatedSkeleton, {
    bottom: 0,
    left: 0,
    position: "absolute",
    right: 0,
    top: 0,
    style: animatedSkeletonStyles
  }));
});
SkeletonLoader.displayName = "SkeletonLoader";
module.exports = __toCommonJS(src_exports);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SkeletonLoader
});
