// @twilio-paste/uid-library/dist/index.debug.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __markAsModule = (target) => __defProp(target, "__esModule", { value: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __reExport = (target, module2, copyDefault, desc) => {
  if (module2 && typeof module2 === "object" || typeof module2 === "function") {
    for (let key of __getOwnPropNames(module2))
      if (!__hasOwnProp.call(target, key) && (copyDefault || key !== "default"))
        __defProp(target, key, { get: () => module2[key], enumerable: !(desc = __getOwnPropDesc(module2, key)) || desc.enumerable });
  }
  return target;
};
var __toESM = (module2, isNodeMode) => {
  return __reExport(__markAsModule(__defProp(module2 != null ? __create(__getProtoOf(module2)) : {}, "default", !isNodeMode && module2 && module2.__esModule ? { get: () => module2.default, enumerable: true } : { value: module2, enumerable: true })), module2);
};
var __toCommonJS = /* @__PURE__ */ ((cache) => {
  return (module2, temp) => {
    return cache && cache.get(module2) || (temp = __reExport(__markAsModule({}), module2, 1), cache && cache.set(module2, temp), temp);
  };
})(typeof WeakMap !== "undefined" ? /* @__PURE__ */ new WeakMap() : 0);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  UIDFork: () => UIDFork,
  uid: () => uid,
  useUID: () => useUID,
  useUIDSeed: () => useUIDSeed
});

// ../../../node_modules/react-uid/dist/es2015/uid.js
var generateUID = function() {
  var counter2 = 1;
  var map = /* @__PURE__ */ new WeakMap();
  var uid2 = function(item, index) {
    if (typeof item === "number" || typeof item === "string") {
      return index ? "idx-" + index : "val-" + item;
    }
    if (!map.has(item)) {
      map.set(item, counter2++);
      return uid2(item);
    }
    return "uid" + map.get(item);
  };
  return uid2;
};
var uid = generateUID();

// ../../../node_modules/react-uid/dist/es2015/Control.js
var React3 = __toESM(require("react"));

// ../../../node_modules/react-uid/dist/es2015/context.js
var React = __toESM(require("react"));
var createSource = function(prefix) {
  if (prefix === void 0) {
    prefix = "";
  }
  return {
    value: 1,
    prefix,
    uid: generateUID()
  };
};
var counter = createSource();
var source = React.createContext(createSource());
var getId = function(source2) {
  return source2.value++;
};
var getPrefix = function(source2) {
  return source2 ? source2.prefix : "";
};

// ../../../node_modules/react-uid/node_modules/tslib/tslib.es6.js
var extendStatics = function(d, b) {
  extendStatics = Object.setPrototypeOf || { __proto__: [] } instanceof Array && function(d2, b2) {
    d2.__proto__ = b2;
  } || function(d2, b2) {
    for (var p in b2)
      if (b2.hasOwnProperty(p))
        d2[p] = b2[p];
  };
  return extendStatics(d, b);
};
function __extends(d, b) {
  extendStatics(d, b);
  function __() {
    this.constructor = d;
  }
  d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
}

// ../../../node_modules/react-uid/dist/es2015/UIDComponent.js
var React2 = __toESM(require("react"));
var prefixId = function(id, prefix, name) {
  var uid2 = prefix + id;
  return String(name ? name(uid2) : uid2);
};
var UID = function(_super) {
  __extends(UID2, _super);
  function UID2() {
    var _this = _super !== null && _super.apply(this, arguments) || this;
    _this.state = {
      quartz: _this.props.idSource || counter,
      prefix: getPrefix(_this.props.idSource),
      id: getId(_this.props.idSource || counter)
    };
    _this.uid = function(item) {
      return prefixId(_this.state.id + "-" + _this.state.quartz.uid(item), _this.state.prefix, _this.props.name);
    };
    return _this;
  }
  UID2.prototype.render = function() {
    var _a = this.props, children = _a.children, name = _a.name;
    var _b = this.state, id = _b.id, prefix = _b.prefix;
    return children(prefixId(id, prefix, name), this.uid);
  };
  return UID2;
}(React2.Component);

// ../../../node_modules/react-uid/dist/es2015/Control.js
var UIDFork = function(_a) {
  var children = _a.children, _b = _a.prefix, prefix = _b === void 0 ? "" : _b;
  return React3.createElement(UIDConsumer, null, function(id) {
    return React3.createElement(source.Provider, { value: createSource(id + "-" + prefix) }, children);
  });
};
var UIDConsumer = function(_a) {
  var name = _a.name, children = _a.children;
  return React3.createElement(source.Consumer, null, function(value) {
    return React3.createElement(UID, { name, idSource: value, children });
  });
};

// ../../../node_modules/react-uid/dist/es2015/hooks.js
var React4 = __toESM(require("react"));
var generateUID2 = function(context) {
  var quartz = context || counter;
  var prefix = getPrefix(quartz);
  var id = getId(quartz);
  var uid2 = prefix + id;
  var gen = function(item) {
    return uid2 + quartz.uid(item);
  };
  return { uid: uid2, gen };
};
var useUIDState = function() {
  if (false) {
    if (!("useContext" in React4)) {
      throw new Error("Hooks API requires React 16.8+");
    }
  }
  return React4.useState(generateUID2(React4.useContext(source)));
};
var useUID = function() {
  var uid2 = useUIDState()[0].uid;
  return uid2;
};
var useUIDSeed = function() {
  var gen = useUIDState()[0].gen;
  return gen;
};
module.exports = __toCommonJS(src_exports);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  UIDFork,
  uid,
  useUID,
  useUIDSeed
});
/*! *****************************************************************************
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the Apache License, Version 2.0 (the "License"); you may not use
this file except in compliance with the License. You may obtain a copy of the
License at http://www.apache.org/licenses/LICENSE-2.0

THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
MERCHANTABLITY OR NON-INFRINGEMENT.

See the Apache Version 2.0 License for specific language governing permissions
and limitations under the License.
***************************************************************************** */
